src/comm.ts

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+"use strict";
+
+import { DisposableDelegate } from 'phosphor-disposable';
+
+import { 
+  IKernel, IKernelFuture, IKernelMessage, IKernelMessageOptions 
+} from './ikernel';
+
+import { createKernelMessage } from './kernel';
+
+import * as utils from './utils';
+
+
+/**
+ * A client side Comm interface.
+ */
+export
+interface IComm {
+  /**
+   * The uuid for the comm channel.
+   *
+   * Read-only
+   */
+  commId: string;
+
+  /** 
+   * The target name for the comm channel.
+   *
+   * Read-only
+   */
+  targetName: string;
+
+  /**
+   * The onClose handler.
+   */
+  onClose: (data?: any) => void;
+
+  /**
+   * The onMsg handler.
+   */
+  onMsg: (data: any) => void;
+
+  /**
+   * Open a comm with optional data.
+   */
+  open(data?: any, metadata?: any): IKernelFuture;
+
+  /**
+   * Send a comm message to the kernel.
+   */
+  send(data: any, metadata?: any, buffers?:(ArrayBuffer | ArrayBufferView)[]): IKernelFuture;
```

Comment:
Hmm, that's no bueno for this situations. Particularly because `autodispose` is being set to `false` on the future. The most likely thing to happen, is that the consumer ignores the return value of sending on a comm, and inadvertently leaks an insane amount of futures.